### PR TITLE
OSDOCS-10163: adds IPv6 config to MicroShift

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
Version(s):
4.17+

Issue:
[OSDOCS-10163](https://issues.redhat.com/browse/OSDOCS-10163)

Link to docs preview:
[IPv6 networking](https://79226--ocpdocs-pr.netlify.app/microshift/latest/microshift_configuring/microshift-nw-ipv6-config.html)
[Networking customization table update](https://79226--ocpdocs-pr.netlify.app/microshift/latest/microshift_networking/microshift-cni.html#microshift-nw-customization-matrix_microshift-about-ovn-k-plugin)
[Using MicroShift config file title change](https://79226--ocpdocs-pr.netlify.app/microshift/latest/microshift_configuring/microshift-using-config-tools.html)

SME review:
- [x] SME has approved this change.

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/microshift-nw-ipv6-concept.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-concept.adoc:9]: There might be a note here. When migrating to dual stack existing pods need to be restarted or else they remain with a single ip family. The same applies to services.

File: modules/microshift-nw-ipv6-concept.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-concept.adoc:13]: What we do not support is switching between ip families. For example, start a cluster in ipv4 and then changing all ip addresses and cidrs to ipv6. We do support up to two values in the cluster and service network config parameters, but that is dual stack only.

File: modules/microshift-nw-ipv6-concept.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-concept.adoc:15]: We do, that is part of the dual stack implementation. This will be included in the ipv6 doc once all the PRs are merged (they are already open).

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:16]: Multus is different. This here is for cluster wide IPv6. Multus can enable it for pods, but not the control plane/services/node/etc.

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:19]: Not sure what you mean by pre-config? The host needs to have IPv6 networking ready: an IPv6 address and IPv6 routes (including the default).

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:26]: Dual stack has a different way of configuring so I would add it in a different section, yes.

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:51]: This one depends. Changing networks after MicroShift has started at least once is not supported (unless migrating to dual stack). So if MicroShift had started before you need a full cleanup before restarting again. This implies application data loss.
Bottom line, the IP family you choose as the first one can not be changed without a cleanup.

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:54]: In this case we may include information such as the networks defined in the node resource:
```bash
$ oc get node -o jsonpath='{.items[].spec.podCIDRs[]}'
fd01::/64
```

Also the pods and services:
```bash
$ oc get pod -A -o wide
NAMESPACE                  NAME                                      READY   STATUS    RESTARTS   AGE   IP                      NODE           NOMINATED NODE   READINESS GATES
kube-system                csi-snapshot-controller-bb7cb654b-rqrt6   1/1     Running   0          65s   fd01:0:0:1::5           microshift-9   <none>           <none>
kube-system                csi-snapshot-webhook-95f475949-nbz8x      1/1     Running   0          61s   fd01:0:0:1::6           microshift-9   <none>           <none>
openshift-dns              dns-default-cjn66                         2/2     Running   0          62s   fd01:0:0:1::9           microshift-9   <none>           <none>
openshift-dns              node-resolver-ppnjb                       1/1     Running   0          63s   2001:db9:ca7:ff::1db8   microshift-9   <none>           <none>
openshift-ingress          router-default-6d97d7b8b6-wdtmg           1/1     Running   0          61s   fd01:0:0:1::8           microshift-9   <none>           <none>
openshift-ovn-kubernetes   ovnkube-master-gfvp5                      4/4     Running   0          63s   2001:db9:ca7:ff::1db8   microshift-9   <none>           <none>
openshift-ovn-kubernetes   ovnkube-node-bnpjh                        1/1     Running   0          63s   2001:db9:ca7:ff::1db8   microshift-9   <none>           <none>
openshift-service-ca       service-ca-5d7bd9db6-j25bd                1/1     Running   0          60s   fd01:0:0:1::4           microshift-9   <none>           <none>
openshift-storage          lvms-operator-656cd9b59b-bwr47            1/1     Running   0          63s   fd01:0:0:1::7           microshift-9   <none>           <none>
openshift-storage          vg-manager-f7dmk                          1/1     Running   0          27s   fd01:0:0:1::a           microshift-9   <none>           <none>

$ oc get svc -A 
NAMESPACE           NAME                            TYPE           CLUSTER-IP   EXTERNAL-IP                                             PORT(S)                      AGE
default             kubernetes                      ClusterIP      fd02::1      <none>                                                  443/TCP                      3m42s
kube-system         csi-snapshot-webhook            ClusterIP      fd02::4c4f   <none>                                                  443/TCP                      3m20s
openshift-dns       dns-default                     ClusterIP      fd02::a      <none>                                                  53/UDP,53/TCP,9154/TCP       2m58s
openshift-ingress   router-default                  LoadBalancer   fd02::f2e6   2001:db9:ca7:ff::1db8,fd01:0:0:1::2,fd02::1:0,fd69::2   80:31133/TCP,443:31996/TCP   2m58s
openshift-ingress   router-internal-default         ClusterIP      fd02::c55e   <none>                                                  80/TCP,443/TCP,1936/TCP      2m58s
openshift-storage   lvms-operator-metrics-service   ClusterIP      fd02::7afb   <none>                                                  443/TCP                      2m58s
openshift-storage   lvms-webhook-service            ClusterIP      fd02::d8dd   <none>                                                  443/TCP                      2m58s
openshift-storage   vg-manager-metrics-service      ClusterIP      fd02::fc1    <none>                                                  443/TCP                      2m58s

```

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:63]: The example above should do

File: modules/microshift-nw-ipv6-dual-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-dual-stack-config.adoc:61]: TODO nodeIPv6

File: modules/microshift-nw-ipv6-single-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-single-stack-config.adoc:42]: Might be worth mentioning that this is only required in systems where IPv4 is also present. If the host is IPv6 only then this is not needed as MicroShift is able to automatically fill it in by itself.

File: modules/microshift-migrating-ipv6-dual-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-migrating-ipv6-dual-stack-config.adoc:9]: You can also convert from IPv6 to IPv6+IPv4, but not sure if this is something any customer would do.

File: modules/microshift-migrating-ipv6-dual-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-migrating-ipv6-dual-stack-config.adoc:12]: Maybe worth mentioning that this is only required for user's workloads, system ones will do automatically by MicroShift.

File: modules/microshift-migrating-ipv6-dual-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-migrating-ipv6-dual-stack-config.adoc:58]: We need to add `nodeIPv6` here as well, this is a subtlety for dual stack.

File: modules/microshift-migrating-ipv6-dual-stack-config.adoc
Comment (pacevedom) [inline modules/microshift-migrating-ipv6-dual-stack-config.adoc:88]: We need to add nodeIPv6 here as well, this is a subtlety for dual stack.

File: modules/microshift-migrating-ipv6-dual-stack-config.adoc
Comment (ShaunaDiaz) [reply modules/microshift-migrating-ipv6-dual-stack-config.adoc:58]: Would it also have a separate SAN?

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (ShaunaDiaz) [reply modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:46]: Yes, that's a good idea. Revised as one step with two parts.

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (pacevedom) [inline modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:77]: Missed this one, sorry. Max value for ipv4 mask must is 24. In ipv6 case its 64.

File: modules/microshift-migrating-ipv6-dual-stack-config.adoc
Comment (pacevedom) [reply modules/microshift-migrating-ipv6-dual-stack-config.adoc]: Yes. This is already looking good in the examples included in the PR :+1:

**Review by pacevedom — APPROVED** [review]
/lgtm
/hold
holding in case you want more reviews, this looks good to me.

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (kcarmichael08) [inline modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:48]:
```suggestion
You must keep the same first entry across restarts and migrations. This is true for any migration: single-to-dual stack, or dual-to-single stack. A complete wipe of the etcd database is required if a change to the first entry is needed. This might result in application data loss and is not supported.
```

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (kcarmichael08) [inline modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:51]: see previous comment about "desired"

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (kcarmichael08) [inline modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:55]: see previous comment about backticks

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (kcarmichael08) [inline modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:148]:
```suggestion
To return to single stack, you can remove the second entry to the networks and return to the single stack that was configured before migrating to dual-stack.
```
Suggest removing hyphen since it's being used as a subject here

**Review by kcarmichael08 — COMMENTED** [review]
This looks great! Had a few minor suggestions per IBM Style.

I was wondering for the section titles, if "single-stack" and "dual-stack" need a subject (for example, "Configuring IPv6 single-stack networking") or would the hyphen be removed if you are using "single stack" as the subject? Searching our existing docs brings up inconsistent results, so please disregard if not applicable.

**Comment by ShaunaDiaz** [comment]
Closing to reset Prow builds to working server. Will reopen.

File: modules/microshift-nw-ipv6-concept.adoc
Comment (opayne1) [inline modules/microshift-nw-ipv6-concept.adoc:13]:
```suggestion
* If you require cluster-wide IPv6 access, such as for the control plane and other services, use the following configuration examples. The {microshift-short} Multus Container Network Interface (CNI) plugin can enable IPv6 for pods.
```

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (opayne1) [inline modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:44]: I think you could just remove desired here based on https://redhat-documentation.github.io/supplementary-style-guide/#_n and it wouldn't change the sentence at all.

**Review by opayne1 — COMMENTED** [review]
Hey Shauna! I noticed 2 small things. Let me know when you have reviewed them and I can merge! 😄

File: modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc
Comment (ShaunaDiaz) [reply modules/microshift-nw-ipv6-dual-stack-migrating-config.adoc:44]: Ah, I missed that one.

**Pull request merged**
